source/extensions/tracers/datadog/agent_http_client.cc

```diff
@@ -54,18 +52,24 @@ datadog::tracing::Expected<void> AgentHTTPClient::post(const URL& url, HeadersSe
   set_headers(writer);
 
   message->body().add(body);
-  ENVOY_LOG(debug, "submitting trace(s) to {} with payload size {}", url.path, body.size());
+  ENVOY_LOG(debug, "submitting data to {} with payload size {}", url.path, body.size());
 
   if (!collector_cluster_.threadLocalCluster().has_value()) {
     ENVOY_LOG(debug, "collector cluster '{}' does not exist", cluster_);
     stats_.reports_skipped_no_cluster_.inc();
     return datadog::tracing::nullopt;
   }
 
+  auto timeout = std::chrono::duration_cast<std::chrono::milliseconds>(deadline -
+                                                                    time_source_.monotonicTime());
+  ENVOY_LOG(debug, "timeout is {} ms", timeout.count());
+
   Http::AsyncClient::Request* request =
       collector_cluster_.threadLocalCluster()->get().httpAsyncClient().send(
           std::move(message), *this,
-          Http::AsyncClient::RequestOptions().setTimeout(std::chrono::milliseconds(1000)));
+          Http::AsyncClient::RequestOptions().setTimeout(
+              std::chrono::duration_cast<std::chrono::milliseconds>(deadline -
+                                                                    time_source_.monotonicTime())));
```

Comment:
I'm digging around in the Envoy source code to see how (if) the `RequestOptions::timeout` is used. The HTTP asyc client implementation uses it to initialize a `NullRouteImpl`, which then is not used directly. A bit mysterious.

Anyway, I was interested to know what happens when the resulting `std::chrono::milliseconds` is negative. What happens then?